### PR TITLE
Remove ContainerStorable.EncodeAsElement

### PR DESCRIPTION
**Pull request by fxamacker**
Updates https://github.com/onflow/atree/issues/292

Currently, EncodeAsElement is used to encode inlined array and map, while Encode is used to encode standalone array and map.

This PR simplifies encoding API by using Encode() to encode both inlined/standalone array/map.

This helps simplify Cadence integration at https://github.com/onflow/cadence/pull/2882.
______

- [ ] Targeted PR against `main` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/atree/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by turbolent — APPROVED** [review]
Nice!

Is this fixing the case we ran into on Friday in Cadence, the encoding of `SomeStorable` was triggering `Encode` instead of `EncodeAsElement`? 
If so, could you please add a test for it?

File: encode.go [hunk 1]
```diff
@@ -28,44 +28,22 @@ import (
...
 func NewEncoder(w io.Writer, encMode cbor.EncMode) *Encoder {
 	streamEncoder := encMode.NewStreamEncoder(w)
 	return &Encoder{
-		Writer:  w,
-		CBOR:    streamEncoder,
-		encMode: encMode,
+		Writer:           w,
+		CBOR:             streamEncoder,
+		encMode:          encMode,
+		inlinedExtraData: newInlinedExtraData(),
```
Comment (turbolent) [inline encode.go:43]: Can this be created lazily, if needed?

**Comment by fxamacker** [comment]
> Is this fixing the case we ran into on Friday in Cadence, the encoding of `SomeStorable` was triggering `Encode` instead of `EncodeAsElement`?  If so, could you please add a test for it?

This was the simplification I mentioned in Friday's meeting as being in progress.  It was annoying to have both functions but there were bigger fish to fry.

Will look into adding tests, etc. to this PR after resolving urgent edge case I found on Sunday (related to `cadence.SomeStorable` if it contains inlined array or map).

**Review by ramtinms — APPROVED** [review]
Looks good to me.

**Comment by fxamacker** [comment]
@turbolent 

>> Is this fixing the case we ran into on Friday in Cadence, the encoding of `SomeStorable` was triggering `Encode` instead of `EncodeAsElement`?  If so, could you please add a test for it?

>This was the simplification I mentioned in Friday's meeting as being in progress.  It was annoying to have both functions but there were bigger fish to fry.
>
>Will look into adding tests, etc. to this PR after resolving urgent edge case I found on Sunday (related to `cadence.SomeStorable` if it contains inlined array or map).

UPDATE:  I made changes to Cadence with commit https://github.com/onflow/cadence/pull/2882/commits/aa087cf0b697711822031aedc8ec000dabf580b4 and it has more detailed description in the commit message.  

Given this, I will add tests for `StorableSlab` with `SomeStorable` (in separate PR) after  you confirm https://github.com/onflow/cadence/pull/2882/commits/aa087cf0b697711822031aedc8ec000dabf580b4 is OK so I can mimic that behavior in Atree tests.
